### PR TITLE
syscall: disable epoll_pwait2() for now on 32bit

### PR DESCRIPTION
**Pull request by poettering**
Alternative to #18973.

### CODE REVIEW

**Comment by bluca** [comment]
CI's unhappy:

../../src/systemd/src/basic/missing_syscall.h:409:7: error: '__LONG_WIDTH__' is not defined, evaluates to 0 [-Werror,-Wundef]
#  if __LONG_WIDTH__ == 32

**Comment by poettering** [comment]
hmm, what? "cc1: all warnings being treated as errors"

does it really make sense to enable that on the CI for stuff we generate via `#warning`?

**Comment by bluca** [comment]
> hmm, what? "cc1: all warnings being treated as errors"
> 
> does it really make sense to enable that on the CI for stuff we generate via `#warning`?

Is there a way to differentiate between manual \#warning and something given by the compiler?

File: src/basic/missing_syscall.h [hunk 1]
```diff
@@ -406,7 +406,13 @@ static inline int missing_epoll_pwait2(
                 const struct timespec *timeout,
                 const sigset_t *sigset) {
 
-#  if defined(__NR_epoll_pwait2) && HAVE_LINUX_TIME_TYPES_H
+#  if __SIZEOF_LONG__ == 4
+// Someone with an interest in 32bit systems, please have a look at this, and figure out why this hangs on 32bit systems.
+// My educated guess: might be because of issues with the __kernel_timespec translation or because of incorrectly sized sigset_t array.
+#    pragma message "epoll_pwait2() appears to be broken on 32bit archs, someone please have a look!"
+        errno = ENOSYS;
+        return -1;
```
Comment (yuwata) [inline src/basic/missing_syscall.h]: Why do you warn all 32bit arch even if `__NR_epoll_pwait2` is not defined? How about the following?
```
#  if defined(__NR_epoll_pwait2) && HAVE_LINUX_TIME_TYPES_H
#    if __SIZEOF_LONG__ == 4
#      pragma message ...
        errno = ENOSYS;
        return -1;
#    else
        (...)
#    endif
#  else
        errno = ENOSYS;
        return -1;
#  endif

**Comment by poettering** [comment]
> Is there a way to differentiate between manual #warning and something given by the compiler?

I downgrade my warning to "info" level which won#t get upgraded to error, so that should be ok. i.e. use "#pragma message" instead of "#warning"

**Comment by poettering** [comment]
made the change @yuwata suggested. ptal.

**Comment by poettering** [comment]
I have the suspicion disabling this on 32bit is not actually necessary, the errno fix is sufficient. i.e. instead of this PR only #18984 should be merged.

**Comment by bluca** [comment]
Closing in favour of https://github.com/systemd/systemd/pull/18984 as discussed

**Comment by yuwata** [comment]
The issue seems not fixed yet. See https://koji.fedoraproject.org/koji/taskinfo?taskID=64059733.

**Comment by yuwata** [comment]
This does not fix the issue on 32bit systems. Closing.

**Pull request closed without merging**